parsl/probe/probe.py

```diff
@@ -0,0 +1,273 @@
+                os.remove('./env.txt')
+
+            except Exception:
+                raise
+
+    def check_executor(self, func, arg, expected):
+
+        # This method can also check the executors. But the uncommented one
+        # is better, bacause it does not involve dfk.
+        #
+        # dfk = parsl.load(self.config)
+        # for label, executor in dfk.executors.items():
+        #     fut = executor.submit(square, None, 2)
+        #     print(fut.result())
+
+        logger.debug('Begin checking the executor(s) in the config')
+        executors = self.config.executors
+        logger.debug(f'There are {len(executors)} executors in the config')
+
+        avail_spec = {'cores': 2, 'memory': 1000, 'disk': 1000}
+        for i, executor in enumerate(executors):
+            logger.debug(f'\tBegin checking #{i+1} executor in the config')
+            executor.provider.script_dir = '.'
+            executor.provider.channel.script_dir = '.'
+            block_id = executor.start()
+            spec = None
+            if type(executor) is WorkQueueExecutor:
+                spec = avail_spec
+            fut = executor.submit(func, spec, arg)
+            try:
+                if fut.result() == expected:
+                    logger.debug(f'\tWhen starting block {block_id}, #{i+1} executor worked properly')
+                else:
+                    logger.error(f'\tWhen starting block {block_id}, #{i+1} executor did NOT work properly')
+            except Exception:
+                raise
+
+
+def start_logger(filename, name='parsl', level=logging.DEBUG, format_string=None):
+    """Add a file handler and a stream handler.
+
+    Args:
+        - filename (string): Name of the file to write logs to
+        - name (string): Logger name
+        - level (logging.LEVEL): Set the logging level.
+        - format_string (string): Set the format string
+
+    Returns:
+       -  None
+    """
+    if format_string is None:
+        format_string = "%(asctime)s.%(msecs)03d %(name)s:%(lineno)d [%(levelname)s]  %(message)s"
+    formatter = logging.Formatter(format_string, datefmt='%Y-%m-%d %H:%M:%S')
+
+    global logger
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.DEBUG)
+
+    file_handler = logging.FileHandler(filename)
+    file_handler.setLevel(level)
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel(level)
+    stream_handler.setFormatter(formatter)
+    logger.addHandler(stream_handler)
+
+
+def cli_run():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--file", required=True,
+                        help="Config file to be tested")
+    parser.add_argument("-d", "--debug", action='store_true',
+                        help="Set logging debug mode")
+    parser.add_argument("-l", "--logdir", default="config_probe_logs",
```

Comment:
Skip the logdir, since we just have a single log file as output. Otherwise, we should try to log to the runinfo directory?